client/src/app/pages/applications/application-assessment/components/application-assessment-wizard/application-assessment-wizard.tsx

```diff
@@ -0,0 +1,351 @@
+        stakeholderGroups: initialAssessment?.stakeholderGroups || [],
+        comments: initialComments,
+        questions: initialQuestions,
+        [SAVE_ACTION_KEY]: SAVE_ACTION_VALUE.SAVE_AS_DRAFT,
+      };
+    }, [initialAssessment]),
+    // defaultValues: {
+    //   stakeholders: initialAssessment?.stakeholders || [],
+    //   stakeholderGroups: initialAssessment?.stakeholderGroups || [],
+    //   comments: initialComments,
+    //   questions: initialQuestions,
+    //   [SAVE_ACTION_KEY]: SAVE_ACTION_VALUE.SAVE_AS_DRAFT,
+    // },
+    mode: "onChange",
+  });
+  const values = methods.getValues();
+
+  console.log("values", values);
+
+  const errors = methods.formState.errors;
+  const isValid = methods.formState.isValid;
+  const isSubmitting = methods.formState.isSubmitting;
+  const isValidating = methods.formState.isValidating;
+
+  const watchAllFields = methods.watch();
+
+  const disableNavigation = !isValid || isSubmitting;
+
+  const isFirstStepValid = () => {
+    const numberOfStakeholdlers = values.stakeholders.length;
+    const numberOfGroups = values.stakeholderGroups.length;
+    return numberOfStakeholdlers + numberOfGroups > 0;
+  };
+
+  const isQuestionValid = (question: Question): boolean => {
+    const questionErrors = errors.questions || {};
+    return !questionErrors[getQuestionFieldName(question, false)];
+  };
+
+  const isCommentValid = (category: QuestionnaireCategory): boolean => {
+    const commentErrors = errors.comments || {};
+    return !commentErrors[getCommentFieldName(category, false)];
+  };
+
+  const questionHasValue = (question: Question): boolean => {
+    const questionValues = values.questions || {};
+    const value = questionValues[getQuestionFieldName(question, false)];
+    return value !== null && value !== undefined;
+  };
+
+  const commentMinLenghtIs1 = (category: QuestionnaireCategory): boolean => {
+    const categoryComments = values.comments || {};
+    const value = categoryComments[getCommentFieldName(category, false)];
+    return value !== null && value !== undefined && value.length > 0;
+  };
+
+  const shouldNextBtnBeEnabled = (category: QuestionnaireCategory): boolean => {
+    return (
+      category.questions.every((question) => isQuestionValid(question)) &&
+      category.questions.every((question) => questionHasValue(question)) &&
+      isCommentValid(category)
+    );
+  };
+
+  const maxCategoryWithData = [...sortedCategories]
+    .reverse()
+    .find((category) => {
+      return (
+        category.questions.some((question) => questionHasValue(question)) ||
+        commentMinLenghtIs1(category)
+      );
+    });
+  const canJumpTo = maxCategoryWithData
+    ? sortedCategories.findIndex((f) => f.id === maxCategoryWithData.id) + 1
+    : 0;
+
+  const onSubmit = (formValues: ApplicationAssessmentWizardValues) => {
+    if (!initialAssessment) {
+      console.log("An assessment must exist in order to save the form");
+      return;
+    }
```

Comment:
The form validation makes this block obsolete.

Reply:
@ibolton336, can you confirm the form cannot be submitted when validation fails ?
In which case this block is not needed anymore.

Reply:
@gildub This check is actually to make sure that the assessment exists so that we have an assessment to patch. Else we return with an error. The assessment is created before this final step in the wizard. 

There is an onInvalid callback to hook into that can be used to handle any validations we need. That is defined on line 180.

Reply:
@ibolton336, I suppose my confusion comes from `initialAssessment` which is a rename of parameter `assessment`. Can't we just keep `assessment` ?